### PR TITLE
add UTF-8 mode for the FTP adapter

### PR DESCRIPTION
**Pull request by TechAlchemistry**
Now the FTP adapter can work with UTF-8 file paths. The UTF-8 mode is disabled by default. 
Example:
```
use League\Flysystem\Filesystem;
use League\Flysystem\Adapter\Ftp as Adapter;

$filesystem = new Filesystem(new Adapter([
    'host' => $host,
    'username' => $user,
    'password' => $password,
    'utf8' => true,         // enable the UTF-8 mode
]));

$filesystem->read('テスト'); // without the UTF-8 mode will throw FileNotFoundException
```

Another way to enable the UTF-8 mode:
```
// before connect to server (any CRUD operations)
// or you must manually reconnect after that
$filesystem->getAdapater()->setUtf8(true); 
```

### CODE REVIEW

File: src/Adapter/Ftp.php [hunk 1]
```diff
@@ -30,6 +30,11 @@ class Ftp extends AbstractFtpAdapter
     protected $recurseManually = false;
 
     /**
+     * @var bool
+     */
+    protected $utf8 = true;
```
Comment (frankdejonge) [inline src/Adapter/Ftp.php]: Can we make the default false? That's the current behaviour. After that I'll merge and add these configurations to the docs.

Comment (RubtsovAV) [inline src/Adapter/Ftp.php]: Done. But this mode should not break a default behavior.

**Comment by RubtsovAV** [comment]
Can you tag the new version, please?

**Comment by RubtsovAV** [comment]
@frankdejonge Can you tag of new version, please?

**Comment by frankdejonge** [comment]
@RubtsovAV 1.0.38 is all yours 👍

**Comment by RubtsovAV** [comment]
@frankdejonge Thank you so much :)

**Comment by chrisooo3** [comment]
After this config 
```yaml
oneup_flysystem:
    adapters:
        ftp_adapter:
            ftp:
                host: ''
                username: ''
                password: ''
                port: 21
                root: '/'
                passive: true
                ssl: true
                timeout: 30
                utf-8: true
```
I am getting following error: 
```
Unrecognized option "utf_8" under "oneup_flysystem.adapters.ftp_adapter.ftp". Available options are "host", "ignorePassiveAddress", "passive", "password", "permPrivate", "permPublic", "port", "recurseManually", "root", "ssl", "systemType", "timeout", "transferMode", "username".
```
How can I set `utf-8` properly?

**Pull request merged**